src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/special_attributes.jl

```diff
@@ -105,3 +105,111 @@ function intersection_form(v::NormalToricVariety)
     end
     return intersection_dict_for_user
 end
+
+
+@doc raw"""
+    chern_class(v::NormalToricVariety, k::Int; check::Bool = true)
+
+Computes the `k`-th Chern class of the tangent bundle of a normal toric variety
+that is both smooth and complete. Since these checks can be computationally
+very demanding, we provide an optional argument `check`. Once set to `false`,
+this method skips those tests.
+
+The implemented algorithm uses proposition 13.1.2 in [CLS11](@cite).
+
+# Examples
+```jldoctest
+julia> F3 = hirzebruch_surface(NormalToricVariety, 3)
+Normal toric variety
+
+julia> chern_class(F3, 0)
+Cohomology class on a normal toric variety given by 1
+
+julia> chern_class(F3, 1, check = false)
+Cohomology class on a normal toric variety given by t1 + x1 + t2 + x2
+
+julia> integrate(chern_class(F3, 2), check = false)
+4
+```
+"""
+function chern_class(v::NormalToricVariety, k::Int; check::Bool = true)
+  # Consistency checks
+  @req k >= 0 "Chern class index must be non-negative"
+  @req k <= dim(v) "Chern class index must not exceed dimension of the toric variety"
+
+  # Check if we can compute the Chern classes for this toric variety
+  if check
+    @req is_smooth(v) && is_complete(v) "The Chern classes of the tangent bundle are only supported for smooth and complete toric varieties"
+  end
+
+  # If thus far, no non-trivial Chern classes have been computed for this toric variety, add an "empty" vector
+  if !has_attribute(v, :chern_classes)
+    cs = Vector{Any}(nothing, dim(v) + 1)
```

Comment:
```suggestion
    cs = Vector{Union{Nothing,TypeOfCohomologyClass}}(nothing, dim(v) + 1)
```
even something like this may help the julia compiler to produce more efficient code